html/pfappserver/root/src/views/Configuration/sources/_store.js

```diff
@@ -64,11 +66,23 @@ const actions = {
       throw err
     })
   },
-  optionsById: ({ commit }, id) => {
+  optionsById: ({ state, commit }, id) => {
     commit('ITEM_REQUEST')
-    return api.itemOptions(id).then(response => {
+    return api.itemOptions(id).then(async response => {
       commit('ITEM_SUCCESS')
-      return response
+      if (state.cache[id]) {
+        const item = await Promise.resolve(state.cache[id]).then(cache => _.cloneDeep(cache))
+        return filterOutLdapOptions(response, item.type)
+      } else {
+        const item = await api.item(id).then(item => {
+          commit('ITEM_REPLACED', item)
+          return _.cloneDeep(item)
+        }).catch((err) => {
+          commit('ITEM_ERROR', err.response)
+          throw err
+        })
```

Comment:
This is basically `getAuthenticationSource`, but I couldn't find a way to reuse it. We should be able to just do `this.getAuthenticationSource` if we use `defineStore` from pinia